### ISSUE TITLE
Generate overwritten Equal method for C# models

### ISSUE THREAD

**Issue by jonaslagoni** [asyncapi/cli#466]
#### Reason/Context
In Modelina for C#, you can make it generate the models with the Equal method overwritten, this should be possible through the CLI.

To do this I suggest we add a flag called `csharpEqual` as a boolean. If true the generator should include the following options:
```  
presets: [
    {
      preset: CSHARP_COMMON_PRESET,
      options: {
        equal: true
      }
    }
  ]
```
Which should be added here: https://github.com/asyncapi/cli/blob/438de0741e9ef29672b6e3af483b90e796903a7d/src/commands/generate/models.ts#L137 as the code example shows: https://github.com/asyncapi/modelina/blob/master/examples/csharp-generate-equals-and-hashcode/index.ts

These will be the generated results: https://github.com/asyncapi/modelina/blob/master/examples/csharp-generate-json-serializer/__snapshots__/index.spec.ts.snap

Remember to also add a test here: https://github.com/asyncapi/cli/blob/438de0741e9ef29672b6e3af483b90e796903a7d/test/commands/generate/models.test.ts#L95

Related documentation: https://github.com/asyncapi/modelina/blob/master/docs/languages/Csharp.md#generate-models-with-equals-and-gethashcode-methods

Feel free to reach out if you have any questions or get stuck!

**Comment by sambhavgupta0705** [comment]
Hi @jonaslagoni I would like to work on this issue.

**Comment by kaushik-rishi** [comment]
Working on this along with #467 
I'm planning to figure both out together.

**Comment by kaushik-rishi** [comment]
@jonaslagoni can we close this issue as well as it's fixed.

**Comment by kaushik-rishi** [comment]
@jonaslagoni this can be closed :smile:

**Comment by jonaslagoni** [comment]
@kaushik-rishi you have to get a codeowner to do it 😄

**Comment by kaushik-rishi** [comment]
Oh didn't know who can I request for then?

**Comment by jonaslagoni** [comment]
You can always find them in the https://github.com/asyncapi/cli/blob/master/CODEOWNERS file 🙂

**Comment by kaushik-rishi** [comment]
@Souvikns This can be closed. It's fixed in #583
